src/plugins/janus_sip.c

```diff
@@ -3953,9 +3968,15 @@ static void *janus_sip_handler(void *data) {
 				goto error;
 			}
 			janus_mutex_unlock(&session->mutex);
-			JANUS_VALIDATE_JSON_OBJECT(root, accept_parameters,
-				error_code, error_cause, TRUE,
-				JANUS_SIP_ERROR_MISSING_ELEMENT, JANUS_SIP_ERROR_INVALID_ELEMENT);
+			if(progress) {
+				JANUS_VALIDATE_JSON_OBJECT(root, progress_parameters,
+					error_code, error_cause, TRUE,
+					JANUS_SIP_ERROR_MISSING_ELEMENT, JANUS_SIP_ERROR_INVALID_ELEMENT);
+			} else {
+				JANUS_VALIDATE_JSON_OBJECT(root, accept_parameters,
+					error_code, error_cause, TRUE,
+					JANUS_SIP_ERROR_MISSING_ELEMENT, JANUS_SIP_ERROR_INVALID_ELEMENT);
+			}
```

Comment:
Heyy, just wanted to give a small update here, I initially tried to do this using a ternary but was getting errors and it took more effort to fix than to rewrite using the if/else.

Regarding the statuses, I am checking the places where`janus_sip_call_status_incall` is used and the `janus_sip_call_is_established()` function, will be working on this in the next few days to try and decide where we should take the new state into account as well.

Reply:
Small update here: you were right, I should have just copy pasted your suggestion and it would have worked. 

However, correct me if I am wrong, but since this `JANUS_VALIDATE_JSON_OBJECT` is defined as a macro, passing in an expression such as `progress ? progress_parameters : accept_parameters` would insert this expression wherever that parameter is used inside the macro, which is an unnecessary overhead. 

The way I have changed it now is that first we compute the desired params using a ternary operator, then pass the result as a parameter to the macro.